Initial draft update alpha diversity chapter

thomazbastiaanssen opened the pull request:
Hi! 

Please find my first draft here. No code changes, only text (& refs)

thomazbastiaanssen committed:
first draft alpha div edits

thomazbastiaanssen committed:
Merge branch 'microbiome:devel' into tfs

thomazbastiaanssen committed:
Merge pull request #1 from thomazbastiaanssen/tfs

Merge first round of edits with fork

antagomir commented on inst/assets/bibliography.bib:
```diff
@@ -1,3 +1,15 @@
+@article{bastiaanssen2023bugs1,
+  title={Bugs as features (part 1): concepts and foundations for the compositional data analysis of the microbiome--gut--brain axis},
+  author={Bastiaanssen, Thomaz FS and Quinn, Thomas P and Loughman, Amy},
+  journal={Nature Mental Health},
```
maybe better to use double curly brackets {{}} to maintain the capitalization?

antagomir commented on inst/assets/bibliography.bib:
```diff
@@ -1,3 +1,15 @@
+  number={12},
+  pages={930--938},
+  doi = {10.1038/s44220-023-00148-3},
+
```
remove this empty row?

antagomir commented on inst/assets/bibliography.bib:
```diff
@@ -1,3 +1,15 @@
+  doi = {10.1038/s44220-023-00148-3},
+
+  year={2023},
+  publisher={Nature Publishing Group US New York}
```
maybe better to use double curly brackets {{}} to maintain the capitalization?

antagomir reviewed (commented):
Seems good!

thomazbastiaanssen committed:
First pass at new structure

thomazbastiaanssen committed:
according to Leo's suggestions

TuomasBorman commented on inst/pages/alpha_diversity.qmd:
```diff
@@ -5,94 +5,55 @@ library(rebook)
-  (sample). The simplest richness index is the number of species
-  observed in a sample (observed richness). Assuming limited sampling
-  from the community, however, this may underestimate the true species
-  richness. Several estimators have been developed to address this,
-  including for instance ACE [@Chao1992] and Chao1 [@Chao1984]
-  indices. Richness estimates do not aim to characterize variations in
-  species abundances.
-  
-Nonparametric richness estimators such as Chao1 and ACE, however, must not be 
-used with amplicon sequence variant (ASV) data. Algorithms that generate ASVs, 
-like DADA2 and Deblur, typically remove singletons, which are essential for 
-these richness calculations. This removal leads to meaningless results. 
-Although ASVs offer higher resolution than operational taxonomic units (OTUs) 
-and are increasingly used, the removal of singletons invalidates the application 
-of Chao1 and ACE. Therefore, alternative alpha diversity metrics that do not 
-depend on singletons or doubletons should be considered, or OTUs could be 
-used specifically for alpha diversity analysis to retain low-abundance taxa.
-Additionally, the inability of denoising algorithms to distinguish true 
-singleton sequences from artifacts further complicates the issue, making 
-traditional richness estimators unsuitable for ASV datasets, which are often 
-standardized for sequencing depth.[@Deng2024]
-
-**Evenness** focuses on the distribution of species abundances, and it
-  can thus complement the number of species. Pielou's evenness is a
-  commonly used index, obtained by normalizing Shannon diversity by
-  (the natural logarithm of) observed richness.
-
-These main classes of alpha diversity are sometimes complemented with
-indices of dominance or rarity:
-
-**Dominance** indices are in general negatively correlated with alpha
-  diversity. A high dominance is obtained when one or a few species have
-  a high share of the total species abundance in the community. Note
-  that dominance indices are generally inversely correlated with other
-  alpha diversity indices.
-
-**Rarity** indices characterize the concentration of species at low
-  abundance.  Prevalence and detection thresholds determine rare
-  species whose total concentration will determine the value of a
-  rarity index.
```
This old version has more detailed description on "diversity", "richness" and "evenness". These are common terms and it might be therefore good idea to go through in this chapter (either in this kind of list or a table)

What was the justification why these are not highlighted in the new version?

thomazbastiaanssen replied:
Agreed, these are key concepts. I now briefly explain them in the text, in the context of Hill's equation as well.

TuomasBorman commented on inst/pages/alpha_diversity.qmd:
```diff
@@ -5,94 +5,55 @@ library(rebook)
+
+![Alpha Diversity metrics are numerically related and can be classified along two axes. Here, we show Hill coefficient on the x-axis and whether the index considers phylogeny on the y-axis](figures/fig_13_1_alphadiv.png) 
+
+
```
Remove extra line break

TuomasBorman reviewed (changes requested):
Looks very good!

You have apparently noticed, but I am not sure if we had said: try to keep lines under 80 characters long. That will help in maintenance and future Bioc submission process

thomazbastiaanssen committed:
According to PR comments:

Added Hill1973 to bibliography
Fixed the 80-line limit
Replaced figure with table
Re-emphasized diversity, richness & evenness, now referring to them in text and table.

thomazbastiaanssen commented on inst/pages/figures/fig_13_1_alphadiv.png:
The .png should likely not be included now that we have the table

TuomasBorman replied:
Yes, please remove it

TuomasBorman replied:
I removed that

TuomasBorman commented on inst/pages/alpha_diversity.qmd:
````diff
@@ -1,98 +1,115 @@
+Second, some alpha diversity metrics are weighed based on phylogeny, like 
+Faith's PD [-@Faith1992] and PhILR [@Silverman2017]. 
+
+```{r, echo=FALSE}
````
Use quarto style

TuomasBorman replied:
I fixed this.

TuomasBorman commented on inst/pages/alpha_diversity.qmd:
```diff
@@ -1,98 +1,115 @@
-# Community diversity {#sec-community-diversity}
+# Alpha Diversity {#sec-alpha-diversity}
```
This needs to be updated everywhere in the book

TuomasBorman replied:
I fixed this.

TuomasBorman commented on inst/pages/alpha_diversity.qmd:
````diff
@@ -1,98 +1,115 @@
+```{r, echo=FALSE}
+library(gt)
+
+#Descriptions
````
Small thing, but use space after # --> consistent style

TuomasBorman replied:
I fixed this.

TuomasBorman commented on inst/pages/alpha_diversity.qmd:
```diff
@@ -1,98 +1,115 @@
+
+table_colnames <- c("phyl_desc", hill_0, hill_1, hill_2)
+
+footnote_md <- md(
+"The equation for general diversity can be defined as follows: 
+   \\${{}^qD = (\\sum_{i=1}^{R}p^q_i})^{\\frac{1}{(1-q)}}\\$,   
+   with *q* = Hill number, *R* = number of features, 
+   *p* = relative feature abundance.")
+
```
For some reason, this does not work for me. I get this error:

Error: ParseError: KaTeX parse error: Unexpected character: '\' at position 50: …frac{1}{(1-q)}}\̲

TuomasBorman committed:
up

TuomasBorman committed:
up

TuomasBorman commented:
I made some changes. The PR looks good. Final things to do:

1. The diversity table is not rendering in my local computer. Can you check that it is correct?
2. Check that every package that is utilized is listed in DESCRIPTION. (I added gt and katex as it seems that they are needed, please double-check)

thomazbastiaanssen committed:
BiocBook::preview(BiocBook::BiocBook('.')) renders the full book sucessfully.

minor edits:
- fixed some duplicate labels for code chunks.
- removed katex from dependencies in DESCRIPTION

thomazbastiaanssen committed:
fixed some duplicate labels for code chunks.

BiocBook::preview(BiocBook::BiocBook('.')) renders the full book sucessfully.

thomazbastiaanssen committed:
Merge branch 'devel' of https://github.com/thomazbastiaanssen/OMA into devel

thomazbastiaanssen committed:
remove gt from DESCRIPTION

thomazbastiaanssen committed:
restored gt - removed by mistake

thomazbastiaanssen commented:
> I made some changes. The PR looks good. Final things to do:
> 
>     1. The diversity table is not rendering in my local computer. Can you check that it is correct?
> 
>     2. Check that every package that is utilized is listed in DESCRIPTION. (I added gt and katex as it seems that they are needed, please double-check)

Many thanks! The book renders using the given call - katex is not necessary, but gt is. 
I'd say it's ready :)

TuomasBorman committed:
Update DESCRIPTION

TuomasBorman committed:
Update DESCRIPTION
